ipu-plugin/ipuplugin/cmd/rootcmd.go

```diff
@@ -254,3 +251,14 @@ func getBridgeController(bridge, bridgeType, ovsCliDir string) (types.BridgeCont
 		return ipuplugin.NewLinuxBridgeController(bridge), types.LinuxBridge
 	}
 }
+
+func getPluginMode() string {
```

Comment:
I don't understand why this is needed - also in case there is an x86 ipu then this function will not work as intended

Reply:
We were told that user should not have to specify this mode during deployment. The "mode" was a required flag and it makes the deployment little bit inconvenient. I guess if we have x86 IPU then we will have reintroduce the flag again unless there's another way to determining we are on IPU.

Reply:
could we just readjust the code?

when daemon calls our VSP Init function, it passes in a "mode"
can we leverage this ?

i.e.
in VSP main, start grpc for Init to listen, when we get the call, do the Init stuff and then start the VSP in the correct mode?

Reply:
ok - should we add this as an explanation in the code ?

Reply:
> could we just readjust the code?
> 
> when daemon calls our VSP Init function, it passes in a "mode" can we leverage this ?
> 
> i.e. in VSP main, start grpc for Init to listen, when we get the call, do the Init stuff and then start the VSP in the correct mode?

It would be too late for us. To handle Init() method we already need to start grpc services. And the grpc services we start depends  on what mode the plugin is running on.

Reply:
> ok - should we add this as an explanation in the code as comment as a guidance for future iterations?

Sure we can add a comment.